ImageKit/ImageKitUploader.swift

```diff
@@ -6,88 +6,217 @@
+                                customMetadata: customMetadata,
                                 progressClosure: progress,
                                 urlConfiguration: urlConfiguration,
+                                uploadPolicy: policy,
                                 completion: { uploadResult in
                                     completion(uploadResult)
-                            })
-                        } else {
-                            completion(Result.failure(IKError.SignatureError.invalidSignatureResponse("Invalid Signature")))
+                                }
+                            )
                         }
-                    case .failure(let error):
-                        completion(Result.failure(error))
+                        _ = videoProcessor.outputFile(input: file, fileName: fileName)
+                        return
+                    }
+                    UploadAPI.upload(
+                        file: fileData,
+                        token: token,
+                        fileName: fileName,
+                        useUniqueFileName: useUniqueFilename,
+                        tags: tags?.joined(separator: ","),
+                        folder: folder,
+                        isPrivateFile: isPrivateFile,
+                        customCoordinates: customCoordinates,
+                        responseFields: responseFields,
+                        extensions: extensions,
+                        webhookUrl: webhookUrl,
+                        overwriteFile: overwriteFile,
+                        overwriteAITags: overwriteAITags,
+                        overwriteTags: overwriteTags,
+                        overwriteCustomMetadata: overwriteCustomMetadata,
+                        customMetadata: customMetadata,
+                        progressClosure: progress,
+                        urlConfiguration: urlConfiguration,
+                        uploadPolicy: policy,
+                        completion: { uploadResult in
+                            completion(uploadResult)
+                        }
+                    )
                 }
-            })
-        } else {
-            fatalError("Public Key / Authentication Endpoint is not defined while initalizing the SDK")
+            }
         }
 
-    }
-
     public func upload(
         file: UIImage,
+        token: String,
         fileName: String,
-        useUniqueFilename: Bool = true,
-        tags: [String] = [],
-        folder: String? = "/",
-        isPrivateFile: Bool? = false,
-        customCoordinates: String? = "",
-        responseFields: String? = "",
-        signatureHeaders: [String: String]? = [String: String](),
+        useUniqueFilename: Bool? = nil,
+        tags: [String]? = nil,
+        folder: String? = nil,
+        isPrivateFile: Bool? = nil,
+        customCoordinates: String? = nil,
+        responseFields: String? = nil,
+        extensions: [[String : Any]]? = nil,
```

Comment:
Is it possible for us to have a proper type for extensions rather than `[["string": Any]]`

Reply:
@ahnv the extensions here are basically JSON objects whose schema can vary across different extensions, so it is not feasible to define a concrete type for all the extensions. the type `[String : Any]` correctly maps from and to a variable JSON structure.